shepherd/shepherd/shepherd.py

```diff
@@ -169,18 +169,32 @@ def dequeue_and_feed_jobs(self, sheep_id: str) -> None:
                 try:
                     self.start_sheep(sheep_id, model.name, model.version)
                 except SheepConfigurationError as sce:
-                    shutil.rmtree(path.join(sheep.sheep_data_root, job_id))
                     error = 'Failed to start sheep for this job ({})'.format(str(sce))
                     logging.error('Sheep `%s` encountered error when processing job `%s`: %s', sheep_id, job_id, error)
-                    error = error.encode()
-                    self.minio.put_object(job_id, ERROR_FILE, BytesIO(error), len(error))
+                    self._job_failed(job_id, error, sheep)
+                    continue
+                except Exception as e:
```

Comment:
As this happens inside a greenlet, we need to be extra careful here. Otherwise, we might end up with a crashed feeding greenlet and a job stuck inside a sheep's throat (actually the bug ).

Reply:
Valid.